src/analyses/variable-sensitivity/constant_array_abstract_object.cpp

```diff
@@ -486,10 +496,27 @@ constant_array_abstract_objectt::visit_sub_elements(
     std::dynamic_pointer_cast<constant_array_abstract_objectt>(
       mutable_clone());
 
-  for(auto &item : result->map)
+  bool modified = false;
+
+  shared_array_mapt::viewt view;
+  result->map.get_view(view);
+
+  for(auto &item : view)
   {
-    item.second=visitor.visit(item.second);
+    auto newval = visitor.visit(item.second);
+    if(newval != item.second)
+    {
+      result->map[item.first]=newval; 
```

Comment:
We could use `result->map.find(item.first, tvt(true))` here to retrieve the element to replace.